src/realm/util/features.h

```diff
@@ -215,6 +217,27 @@
 #define REALM_TVOS 0
 #endif
 
+// asl_log is deprecated in favor of os_log as of the following versions:
+// macos(10.12), ios(10.0), watchos(3.0), tvos(10.0)
+// versions are defined in /usr/include/Availability.h as they become available
+#if REALM_PLATFORM_APPLE \
+   && ( (defined(MAC_OS_X_VERSION_10_12) \
+         && defined(MAC_OS_X_VERSION_MIN_REQUIRED) \
+         && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12) \
```

Comment:
Thanks for pointing that out, I must have been mixing up the docs I was reading. I'll change it to use the macros from Availability.h.

I like the pattern you suggested, but unfortunately we still need to check if the `xx_VERSION_MIN_REQUIRED` is defined because we compile with `-Wundef`: it will evaluate to 0 if undefined but it will also produce a warning.